### PR TITLE
feat: update google-drive sync template to support more file formats

### LINKED ISSUE #2661: `Google Drive` sync does not take into account Google Sheets files
When adding folders that contains both Google Docs and Google Sheets files the sync only works for Google Docs. 

**WEBHOK FROM NANGO**:

```
{
  "syncName": "documents",
  "modifiedAfter": "2024-09-01T18:37:34.539Z",
  "success": true,
  "connectionId": "219ec444-e8f5-41c6-bca7-e09ee5cc3e5a",
  "from": "nango",
  "model": "Document",
  "queryTimeStamp": "2024-09-01T18:37:34.539Z",
  "type": "sync",
  "syncType": "INITIAL",
  "providerConfigKey": "google-drive-pjdc",
  "responseResults": {
    "deleted": 0,
    "added": 1,
    "updated": 0
  }
}
```

**FOLDER CONTENT**:

```
{
    "kind": "drive#fileList",
    "incompleteSearch": false,
    "files": [
        {
            "kind": "drive#file",
            "mimeType": "application/vnd.google-apps.spreadsheet",
            "id": "14AgIrn21IRqjWQ3JTm_YM40qNVcNqJvPi406ln1gN5A",
            "name": "Characters GOT"
        },
        {
            "kind": "drive#file",
            "mimeType": "application/vnd.google-apps.folder",
            "id": "1rtARbNcjBnB2BbZZ-2Kj5xn9NeaQSVHt",
            "name": "Characters"
        },
        {
            "kind": "drive#file",
            "mimeType": "application/vnd.google-apps.document",
            "id": "1_i3TF66gZ-FbwFUA459X9Arn0shGiSMKgNBzWPPMkrQ",
            "name": "Introduction GOT"
        }
    ]
}
```

### PR DESCRIPTION
**Pull request by homanp**
## Describe your changes

Fixes #2661 

## Checklist before requesting a review (skip if just adding/editing APIs & templates)
- [ ] I added tests, otherwise the reason is: 
- [ ] I added observability, otherwise the reason is:
- [ ] I added analytics, otherwise the reason is: 

## Summary by CodeRabbit

- **New Features**
	- Enhanced file synchronization with Google Drive by introducing a refined filtering process for supported MIME types.
	- Improved error handling for unsupported file types to ensure smoother processing.
  
- **Bug Fixes**
	- Resolved issues related to metadata handling and file processing logic, ensuring only acceptable file types are processed.

### CODE REVIEW

File: integration-templates/google-drive/syncs/documents.ts [hunk 1]
```diff
@@ -102,30 +134,31 @@ export default async function fetchData(nango: NangoSync): Promise<void> {
...
+    mimeType: string
+): Promise<string | null> {
     try {
-        if (mimeType === 'application/vnd.google-apps.spreadsheet') {
+        if (structuredMimeTypeMapping[mimeType]) {
             const contentResponse = await nango.get({
                 endpoint: `drive/v3/files/${doc.id}/export`,
                 params: {
-                    mimeType: 'text/csv'
+                    mimeType: "text/csv",
```
Comment (TBonnin) [inline integration-templates/google-drive/syncs/documents.ts]: should it not be `mimeType: structuredMimeTypeMapping[mimeType]`

**Review by TBonnin — COMMENTED** [review]
thank you for your contribution @homanp 
Can you revert the formatting changes please? Otherwise it looks good, I only made one comment

File: integration-templates/google-drive/syncs/documents.ts [hunk 2]
```diff
@@ -38,30 +52,37 @@ export default async function fetchData(nango: NangoSync): Promise<void> {
...
+                } else if (whiteListedMimeTypes.has(file.mimeType)) {
+                    const content = await fetchDocumentContent(
+                        nango,
+                        file,
+                        file.mimeType
+                    );
                     batch.push({
                         id: file.id,
                         url: file.webViewLink,
-                        content: content || '',
```
Comment (bastienbeurier) [inline integration-templates/google-drive/syncs/documents.ts:59]: This sync seems to save the document's content in the Nango cache, which is not meant to store rich content. 

The typical pattern is that only the document metadata is stored (with `batchSave`), then the document's content is downloaded from the developer app using the document metadata fetched from Nango. 

cc @bodinsamuel @TBonnin for visibility

Comment (homanp) [reply integration-templates/google-drive/syncs/documents.ts:59]: @bastienbeurier had a discussion with Kelvin Wari about this. Have updated the code based on our discussions.

**Comment by khaliqgant** [comment]
@hassan254-prog what should happen with this PR? I see you made a pr that referenced this one? Did that incorporate these changes?

**Comment by hassan254-prog** [comment]
> @hassan254-prog what should happen with this PR? I see you made a pr that referenced this one? Did that incorporate these changes?

Yes, the merged PR incorporated the changes. We now have a sync that fetches the metadata and an action that fetches the content of the document.

**Comment by khaliqgant** [comment]
> > @hassan254-prog what should happen with this PR? I see you made a pr that referenced this one? Did that incorporate these changes?
> 
> Yes, the merged PR incorporated the changes. We now have a sync that fetches the metadata and an action that fetches the content of the document.

Thanks, closing this!

**Pull request closed without merging**